### PR TITLE
Provenance 

### PR DESCRIPTION
**Pull request by ddecouteau**
Adds provenance when creating patient-privacy, advance directives, and Portable Medical Orders consent resources.  Adds baseline (basic) Questionnaire for patient-privacy, implements QuestionnaireResponse for patient-privacy.  Adds english and spanish properties for processing questionnaire response for patient-privacy consent.

### CODE REVIEW

File: leap-consent-ui/src/main/java/gov/hhs/onc/leap/backend/fhir/client/utils/FHIRProvenance.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+@Service
+public class FHIRProvenance {
+    private String PROVENANCE_AGENT = "LEAP Consent Management Service";
+    private String PROVENANCE_AGENT_SYSTEM = "http://terminology.hl7.org/CodeSystem/provenance-participant-type";
+    private String PROVENANCE_AGENT_CODE = "assembler";
+    private String PROVENANCE_AGENT_DISPLAY = "Assember";
+
+    private String PROVENANCE_ACTIVITY_SYSTEM = "http://terminology.hl7.org/CodeSystem/v3-DataOperation";
+    private String PROVENANCE_ACTIVITY_CODE= "CREATE";
+    private String PROVENANCE_ACTIVITY_DISPLAY = "create";
```
Comment (sgroh) [inline leap-consent-ui/src/main/java/gov/hhs/onc/leap/backend/fhir/client/utils/FHIRProvenance.java:20]: These must be final and can be also static (if we change the visibility)

File: leap-consent-ui/src/main/resources/vaadinapp_es.properties [hunk 2]
```diff
@@ -864,3 +864,29 @@ UserPreferences-emergency_contact=Contacto de Emergencia
...
+PatientPrivacy-questionnaire_response_item_3_2=Organización
+PatientPrivacy-questionnaire_response_item_3_2_1=Organización - Custodio (Seleccionar de la lista).
+PatientPrivacy-questionnaire_response_item_4_1=Facultativo
+PatientPrivacy-questionnaire_response_item_4_1_1=Practicante - Destinatario (Seleccionar de la lista).
+PatientPrivacy-questionnaire_response_item_4_2=Organización
+PatientPrivacy-questionnaire_response_item_4_2_1=Organización - Destinatario (Seleccionar de la lista).
+PatientPrivacy-questionnaire_response_item_5_1=Quitarlos
+PatientPrivacy-questionnaire_response_item_5_2=No tengo preocupaciones de privacidad
+PatientPrivacy-questionnaire_response_item_5_1_1=Seleccionar todo: esto eliminará toda la información marcada como confidencial.
+PatientPrivacy-questionnaire_response_item_6=Este último paso capturará su firma y creará un pdf legible por humanos de este consentimiento. (Firma electrónica capturada)
```
Comment (sgroh) [inline leap-consent-ui/src/main/resources/vaadinapp_es.properties:892]: Please change this to: 
Este último paso capturará su firma y creará un pdf legible de este consentimiento. (Firma electrónica capturada)

**Review by sgroh — CHANGES_REQUESTED** [review]

File: leap-consent-ui/src/main/java/gov/hhs/onc/leap/backend/fhir/client/utils/FHIRProvenance.java (hunk 1 above)
Comment (ddecouteau) [reply leap-consent-ui/src/main/java/gov/hhs/onc/leap/backend/fhir/client/utils/FHIRProvenance.java:20]: thanks done

File: leap-consent-ui/src/main/resources/vaadinapp_es.properties (hunk 2 above)
Comment (ddecouteau) [reply leap-consent-ui/src/main/resources/vaadinapp_es.properties:892]: thanks done

**Review by ddecouteau — COMMENTED** [review]
Changes completed

**Comment by mojitoj** [comment]
@ddecouteau this looks great and obviously good to go. 
One suggestion I have is to have the Provenance resource template as a json or XML file and read/parse that in and only set the attributes that are context/transaction dependent. This would keep the code shorter and also make it easier to change the template --without having to touch a lot of code.

**Pull request merged**